WebRoot/h/changepass

```diff
@@ -445,12 +492,52 @@
                 errorMessageDiv.style.display = "none";
             }
 
-            setSaveButtonDisabled(enabledRules.length > matchedRule.length || confirmPasswordInput.value !== newPasswordInput.value);
+            handleConfirmPasswordChange();
         };
 
         function handleConfirmPasswordChange() {
             resetImg(confirmPasswordInput.value === newPasswordInput.value, getElement("mustMatchCheckImg"), getElement("mustMatchCloseImg"));
-            setSaveButtonDisabled(confirmPasswordInput.value !== newPasswordInput.value);
+            setSaveButtonDisabled();
+        };
+
+        function dryRun() {
+            matchedRemoteRules = [];
+            var request = {
+                Body: {
+                    ChangePasswordRequest:{
+                        account: {
+                            by : "name",
+                            _content : "${mailbox.accountInfo.name}"
+                        },
+                        oldPassword: getElement("oldPassword").value,
+                        password: newPasswordInput.value,
+                        _jsns: "urn:zimbraAccount",
+                        dryRun: "1"
+                    }
+                },
+                Header: {
+                    context: {
+                        _jsns: "urn:zimbra",
+                        csrfToken: "${mailbox.csrfToken}"
+                    }
+                }
+            };
+            var xmlhttp=new XMLHttpRequest();
+            xmlhttp.open("POST","/service/soap/ChangePasswordRequest",false);
+            xmlhttp.send(JSON.stringify(request));
+
+            if (xmlhttp.status == 200 || xmlhttp.status == 201) {
+                var jsonResponse = JSON.parse(xmlhttp.responseText);
```

Comment:
JSON.parse and stringify should be wrapped in try/catch blocks to handle runtime exceptions

Reply:
Fixed.